fix/stuck-instruments.lua

```diff
@@ -0,0 +1,65 @@
+-- Fixes instruments that never got played during a performance
+
+local help = [====[
+
+fix/stuck-instruments
+=====================
+
+Fixes instruments that were picked up for a performance, but were instead
+simulated and are now stuck permanently in a job that no longer exists.
+
+This works around the issue encountered with :bug:`9485`, and should be run
+if you notice any instruments lying on the ground that seem to be stuck in a
+job.
+
+Run ``fix/stuck-instruments -n`` or ``fix/stuck-instruments --dry-run`` to
+list how many instruments would be fixed without performing the action.
+
+]====]
+
+
+function fixInstruments(args)
+    local dry_run = false
+    local fixed = 0
+    for _, arg in pairs(args) do
+        if args[1]:match('-h') or args[1]:match('help') then
+            print(help)
+            return
+        elseif args[1]:match('-n') or args[1]:match('dry') then
+            dry_run = true
+        end
+    end
+    for _, item in ipairs(df.global.world.items.all) do
+        if item:getType() == df.item_type.INSTRUMENT then
```

Comment:
instead of putting all the logic in an if block, consider using the lua version of `continue`:
```lua
for _,item in ipairs(df.global.world.items.all) do
    if item:getType() ~= df.item_type.INSTRUMENT then
        goto continue
    end
    -- ... other logic
    ::continue::
end

Reply:
however, this particular suggestion is moot since you could iterate through `df.global.world.items.other.INSTRUMENT` and get the items you want directly. There is also `df.global.world.items.other.INSTRUMENT_STATIONARY`, but you're not interested in stationary instruments, right?

Reply:
oh, you can use the `::continue::` idiom below where you check for the general ref type